src/appleseed/renderer/modeling/bssrdf/separablebssrdf.cpp

```diff
@@ -246,11 +247,23 @@ namespace
             probe_ray.m_tmin = 1.0e-6;
             probe_ray.m_tmax = norm(exit_point - probe_ray.m_org);
 
-            // Only consider incoming points on the same object instance,
-            // and with the same material as the outgoing point.
-            if (incoming_point.get_object_instance().get_uid() == outgoing_obj_inst_uid &&
-                (incoming_point.get_material() == outgoing_material ||
-                 incoming_point.get_opposite_material() == outgoing_material))
+            const Material* incoming_material = incoming_point.get_material();
+            const BSSRDF* incoming_bssrdf =
+                incoming_material == NULL ? NULL :
+                incoming_material->get_uncached_bssrdf();
+            const Material* incoming_opposite_material = incoming_point.get_opposite_material();
+            const BSSRDF* incoming_opposite_bssrdf =
+                incoming_opposite_material == NULL ? NULL :
+                incoming_opposite_material->get_uncached_bssrdf();
```

Comment:
As explained in the review of your previous PR, you must not call the uncached variants of these methods.